src/App.tsx

```diff
@@ -14,11 +15,74 @@ const goodsFromServer: string[] = [
   'Garlic',
 ];
 
-const App: React.FC = () => (
-  <div className="App">
-    <h1>Selected good: -</h1>
-    {goodsFromServer.length}
-  </div>
-);
+type State = {
+  selectedGood: string | null;
```

Comment:
Actually i think that blank string will return false when converted to boolean, so you don't really need null here